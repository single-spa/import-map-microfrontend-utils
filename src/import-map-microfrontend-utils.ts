export function microfrontendNameToFolderName(
  microfrontendName: string,
): string {
  return microfrontendName.replace(/@/g, "").replace(/\//, "-");
}
