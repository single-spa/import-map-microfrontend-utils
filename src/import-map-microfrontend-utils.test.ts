import { describe, expect, it } from "@jest/globals";
import { microfrontendNameToFolderName } from "./import-map-microfrontend-utils";

describe(`microfrontendNameToFolderName`, () => {
  it(`can process unscoped packages`, () => {
    expect(microfrontendNameToFolderName("root-config")).toMatchSnapshot();
    expect(microfrontendNameToFolderName("navbar")).toMatchSnapshot();
    expect(microfrontendNameToFolderName("left_nav")).toMatchSnapshot();
  });

  it(`can process scoped packages`, () => {
    expect(microfrontendNameToFolderName("@org/root-config")).toMatchSnapshot();
    expect(microfrontendNameToFolderName("@org-name/navbar")).toMatchSnapshot();
    expect(
      microfrontendNameToFolderName("@org_name/left_nav"),
    ).toMatchSnapshot();
  });
});
