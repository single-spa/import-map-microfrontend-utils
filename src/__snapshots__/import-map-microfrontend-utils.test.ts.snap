// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`microfrontendNameToFolderName can process scoped packages 1`] = `"org-root-config"`;

exports[`microfrontendNameToFolderName can process scoped packages 2`] = `"org-name-navbar"`;

exports[`microfrontendNameToFolderName can process scoped packages 3`] = `"org_name-left_nav"`;

exports[`microfrontendNameToFolderName can process unscoped packages 1`] = `"root-config"`;

exports[`microfrontendNameToFolderName can process unscoped packages 2`] = `"navbar"`;

exports[`microfrontendNameToFolderName can process unscoped packages 3`] = `"left_nav"`;
